File: hw/maisincho/u3/HW19AbstractFactory/AbstractFactoryPy/controller/GUIFactory.py
import platform
from abc import ABC, abstractmethod

class GUIFactory(ABC):
    @staticmethod
    def get_factory():
        from controller.WinFactory import WinFactory
        from controller.LinuxFactory import LinuxFactory

        sys = GUIFactory.__read_from_config_file("OS_TYPE")
        if sys == 0:
            return WinFactory()
        else:
            return LinuxFactory()

    @staticmethod
    def __read_from_config_file(key):
        os_name = platform.system().lower()
        if "win" in os_name:
            return 0
        else:
            return 1

    @abstractmethod
    def create_button(self):
        pass

    @abstractmethod
    def create_menu(self):
        pass

File: hw/unda/U3/HW18AbstractFactory/GUIFactoryCorrect/src/main/java/ec/edu/espe/guifactory/controller/GUIFactory.java
package ec.edu.espe.guifactory.controller;

import ec.edu.espe.guifactory.model.Button;
import ec.edu.espe.guifactory.model.LinuxFactory;
import ec.edu.espe.guifactory.model.Menu;
import ec.edu.espe.guifactory.model.WinFactory;

/**
 *
 * @author Mateo Unda
 */
public abstract class GUIFactory {
    public static GUIFactory getFactory() {
        int sys = readFromConfigFile("OS_TYPE");
    if (sys == 0) {
            return (new WinFactory());
        } else {
            return (new LinuxFactory());
        }
    }

    private static int readFromConfigFile(String os_type) {
        System.out.println("ReadFile");
        return 0;
    }

    public abstract Button createButton();

    public abstract Menu createMenu();
}
